### ISSUE TITLE
Unwrap variable only from complex string

### ISSUE THREAD

**Issue by mvorisek** [PHP-CS-Fixer/PHP-CS-Fixer#7766]
## Feature request

Example:

```diff
-$a = "{$msg}";
+$a = $msg;
-$a = "{$obj->msg}";
+$a = $obj->msg;
$a = "{$msg} "; // do nothing, complex string consists of more than a variable
```

**Comment by julienfalque** [comment]
I think this should use the `(string)` cast to keep the same behavior:
```diff
-$a = "{$msg}";
+$a = (string) $msg;
-$a = "{$obj->msg}";
+$a = (string) $obj->msg;
$a = "{$msg} "; // do nothing, complex string consists of more than a variable
```

Why would this be limited to complex strings though? Couldn't the fixer also fix
```diff
-$a = "$msg";
+$a = (string) $msg;
```
?

> should this be part of [`single_quote`](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/blob/v3.48.0/doc/rules/string_notation/single_quote.rst) fixer?

I don't think so, this looks like an entirely new feature to me and I would create a new fixer.

**Comment by mvorisek** [comment]
Good point with the cast. I did not realized it when I was opening this issue. I probably do not want to pursue this issue then and simply find and fix issues in our codebase using `["']\{.+\}["']` regex.
